h/form.py

```diff
@@ -46,6 +49,10 @@ def __call__(self, template_name, **kwargs):
         context = self._system.copy()
         context.update(kwargs)
 
+        context.setdefault(
+            "get_csrf_token", partial(get_csrf_token, context["request"])
+        )
```

Comment:
Adding Pyramid's `get_csrf_token()` method to the Jinja2 environment that's used to render our custom Deform templates, so that the base template for forms can automatically add a CSRF token.

TODO: There are lots of templates throughout the codebase that do their own `get_csrf_token()` calls, it should be possible to remove these now.

Reply:
> TODO: There are lots of templates throughout the codebase that do their own get_csrf_token() calls, it should be possible to remove these now. Actually they may always have been unnecessary? Not sure.

I had a look through and I think they're all actually necessary: we have lots of forms that're just in Jinja templates, not actually rendered by Deform.